r/tests/testthat/test-dplyr-filter.R

```diff
@@ -155,6 +155,59 @@ test_that("filter() with %in%", {
   )
 })
 
+test_that("filter() with between()", {
+  expect_dplyr_equal(
+    input %>%
+      filter(between(dbl, 1, 2)) %>%
+      collect(),
+    tbl
+  )
+
+  expect_dplyr_equal(
+    input %>%
+      filter(between(dbl, 0.5, 2)) %>%
+      collect(),
+    tbl
+  )
+
+  expect_dplyr_equal(
```

Comment:
This is your failing test: because `dplyr` doesn't support the vector-wise comparison. You can rewrite it without `expect_dplyr_equal`, just do something like:

```
expect_identical(
  tbl %>%
    record_batch() %>%
    filter(between(dbl, int, dbl2)) %>%
    collect(),
  tbl %>%
    filter(dbl >= int, dbl <= dbl2)
)
```

or something like that.

Reply:
Looks like fixing that let the next tests run, and some of them fail too.

Reply:
You should just use regular `expect_error` here since the arrow version behaves slightly differently (even when the dplyr version and this version both error, they give different validation messages). `expect_dplyr_error` asserts that they give the exact same error message: https://github.com/apache/arrow/blob/master/r/tests/testthat/helper-expectation.R#L98-L124

Reply:
Ok tests passing now. A few iterations to figure out your test suite 🤕 .